spec/features/availability_spec.rb

```diff
@@ -0,0 +1,191 @@
+    end
+
+    it 'that has holdings available in exactly 2 locations' do
+      visit '/?utf8=✓&search_field=all_fields&q=0060125896'
+      expect(page).to have_selector '.availability-snippet',
+                                    exact_text: 'Altoona, Pattee Library and Paterno Library Stacks'
+    end
+
+    it 'that has holdings available in only 1 location' do
+      visit '/?utf8=✓&search_field=all_fields&q=9780147513861'
+      expect(page).to have_selector '.availability-snippet',
+                                    exact_text: 'Pattee Library and Paterno Library Stacks'
+    end
+
+    it 'and clicks the \'View Availability\' button to display and hide holdings' do
+      visit '/?utf8=✓&search_field=all_fields&q=9781599901091'
+      expect(page).to have_selector 'button[data-target="#availability-5112336"]'
+      expect(page).not_to have_selector '.availability-5112336'
+      click_button('View Availability')
+      expect(page).not_to have_selector '.availability-5112336', wait: 3
+      expect(page).to have_content 'PZ8.G3295Su 2008'
+      expect(page).to have_content 'Fiction G4672sun 2008'
+    end
+
+    it 'that is an online resource and has no holdings to display' do
+      visit '/?utf8=✓&search_field=all_fields&q=D-humanos+Arruti%2C+Mariana'
+      expect(page).not_to have_selector('button[data-target="#availability-22091400"]')
+      expect(page).not_to have_selector '.availability-22091400'
+    end
+  end
+
+  describe 'User opens the item page for a record' do
+    it 'that has holdings to display', js: true do
+      visit '/catalog/1839879'
+      expect(page).to have_selector 'div[class="availability"][data-keys="1839879"]'
+    end
+
+    it 'that is an online resource and has no holdings to display', js: true do
+      visit '/catalog/22091400'
+      expect(page).not_to have_selector 'div[class="availability"]'
+    end
+  end
+
+  describe 'User opens a record with many holdings in a library' do
+    it 'toggles 4 or more holdings with View More/View Less button', js: true do
+      visit '/catalog/1793712'
+      within 'div[data-library="UP-MICRO"]' do
+        expect(page).to have_selector 'button', text: /View More/
+        expect(page).not_to have_selector 'button', text: /View Less/
+        expect(page).to have_xpath './/tbody/tr', count: 4
+        click_button('View More')
+        sleep 1 # let collapse animation finish and wait for it to re-collapse
+        expect(page).to have_xpath './/tbody/tr', count: 28
+        expect(page).to have_selector 'button', text: /View Less/
+        click_button('View Less')
+        expect(page).to have_xpath './/tbody/tr', count: 4
+      end
+    end
+  end
+
+  describe 'Archival Material:' do
+    it 'has \'Request Material\' link so it can be requested through Aeon', js: true do
+      visit '/catalog/1836205'
+      expect(page).to have_css 'a[data-type="aeon-link"][data-call-number="RBM 2457 box1 AX/SP/10155/11"]'
```

Comment:
This failed occasionally for me locally, but I'm not sure anything can be done about this without significant effort.